server/jetstream_api.go

```diff
@@ -1020,6 +1024,55 @@ func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, subject, re
 		return
 	}
 
+	// Do some pre-checking for mirror config to avoid cycles in clustered mode.
+	if cfg.Mirror != nil {
+		if len(cfg.Subjects) > 0 {
+			resp.Error = &ApiError{Code: 400, Description: "stream mirrors can not also contain subjects"}
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
+		if len(cfg.Sources) > 0 {
+			resp.Error = &ApiError{Code: 400, Description: "stream mirrors can not also contain other sources"}
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
+		if cfg.Mirror.FilterSubject != _EMPTY_ {
+			resp.Error = &ApiError{Code: 400, Description: "stream mirrors can not contain filtered subjects"}
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
+		if cfg.Mirror.OptStartSeq > 0 && cfg.Mirror.OptStartTime != nil {
+			resp.Error = &ApiError{Code: 400, Description: "stream mirrors can not have both start seq and start time configured"}
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
```

Comment:
hmm, so then stream mirrors can in fact have no config at all, so the *StreamSource only contributes Name at that point, are we keeping *StreamSource there as a nod to the future when we somehow point to another account in there too?

Your 4 checks here - no subjects, no sources, no filter and no start seq or time - effectively makes the cfg.Mirror only use Name nothing else - since everything else errors.

So why does Mirrors have it? So that we can later add things to point to a stream in another account?

Reply:
Sorry, My bad, dont read code at 9pm

Reply:
Just quick check if you accidentally set both start seq and start time.